pkg/bpf/map_linux_test.go

```diff
@@ -87,16 +113,270 @@ func (s *BPFPrivilegedTestSuite) TestOpen(c *C) {
+	c.Check(err, Not(IsNil))
+	c.Check(value, IsNil)
+
+	value, err = parallelMap.Lookup(key1)
+	c.Check(err, Not(IsNil))
+	c.Check(value, IsNil)
+	value, err = parallelMap.Lookup(key2)
+	c.Check(err, IsNil)
+	c.Check(value, DeepEquals, value2)
+
+	err = parallelMap.EndParallelMode()
+	c.Check(err, IsNil)
+	err = parallelMap.EndParallelMode()
+	c.Check(err, Not(IsNil))
+}
+
+func (s *BPFPrivilegedTestSuite) TestBasicManipulation(c *C) {
+	// existingMap is the same as testMap. Opening should succeed.
+	existingMap := NewMap("cilium_test",
+		MapTypeHash,
+		int(unsafe.Sizeof(TestKey{})),
+		int(unsafe.Sizeof(TestValue{})),
+		maxEntries,
+		BPF_F_NO_PREALLOC,
+		0,
+		DumpParserFunc).WithCache()
+	err := existingMap.Open()
+	c.Check(err, IsNil) // Avoid assert to ensure Close() is called below.
+
+	key1 := &TestKey{Key: 103}
+	value1 := &TestValue{Value: 203}
+	key2 := &TestKey{Key: 104}
+	value2 := &TestValue{Value: 204}
+
+	err = existingMap.Update(key1, value1)
+	c.Check(err, IsNil)
+	// key    val
+	// 100    200
+	value, err := existingMap.Lookup(key1)
+	c.Check(err, IsNil)
+	c.Check(value, DeepEquals, value1)
+	value, err = existingMap.Lookup(key2)
+	c.Check(err, Not(IsNil))
+	c.Check(value, Equals, nil)
+
+	err = existingMap.Update(key1, value2)
+	c.Check(err, IsNil)
+	// key    val
+	// 100    201
+	value, err = existingMap.Lookup(key1)
+	c.Check(err, IsNil)
+	c.Check(value, DeepEquals, value2)
+
+	err = existingMap.Update(key2, value2)
+	c.Check(err, IsNil)
+	// key    val
+	// 100    201
+	// 101    201
+	value, err = existingMap.Lookup(key1)
+	c.Check(err, IsNil)
+	c.Check(value, DeepEquals, value2)
+	value, err = existingMap.Lookup(key2)
+	c.Check(err, IsNil)
+	c.Check(value, DeepEquals, value2)
+
+	err = existingMap.Delete(key1)
+	c.Check(err, IsNil)
+	// key    val
+	// 101    201
+	value, err = existingMap.Lookup(key1)
+	c.Check(err, Not(IsNil))
+	c.Check(value, Equals, nil)
+
+	err = existingMap.DeleteAll()
+	c.Check(err, IsNil)
+	value, err = existingMap.Lookup(key1)
+	c.Check(err, Not(IsNil))
+	c.Check(value, Equals, nil)
+	err = existingMap.DeleteAll()
+	c.Check(err, IsNil)
+
+	err = existingMap.Close()
+	c.Assert(err, IsNil)
+}
+
+func (s *BPFPrivilegedTestSuite) TestDump(c *C) {
+	key1 := &TestKey{Key: 105}
+	value1 := &TestValue{Value: 205}
+	key2 := &TestKey{Key: 106}
+	value2 := &TestValue{Value: 206}
+
+	err := testMap.Update(key1, value1)
+	c.Assert(err, IsNil)
+	err = testMap.Update(key2, value1)
+	c.Assert(err, IsNil)
+	err = testMap.Update(key2, value2)
+	c.Assert(err, IsNil)
+
+	dump1 := map[string][]string{}
+	testMap.Dump(dump1)
+	c.Check(dump1, DeepEquals, map[string][]string{
+		"key=105": []string{"value=205"},
+		"key=106": []string{"value=206"},
+	})
+
+	dump2 := map[string][]string{}
+	customCb := func(key MapKey, value MapValue) {
+		dump2[key.String()] = append(dump2[key.String()], "custom-"+value.String())
+	}
+	testMap.DumpWithCallback(customCb)
+	c.Check(dump2, DeepEquals, map[string][]string{
```

Comment:
If you use `checker.DeepEquals` (from `pkg/datapath/checker`) for cases like this, then it will format errors nicely which makes debug a bit easier when there's any problem. Would you mind updating these?

Reply:
Ok I'll fix these. Thanks!

Reply:
Done (https://github.com/cilium/cilium/pull/7538/commits/8280912414b4b50800a879c2582e80a4fe08eb54)